kernel/classes/datatypes/ezimage/ezimagealiashandler.php

```diff
@@ -378,13 +378,36 @@ function imagePathByNode( $contentObjectAttribute, $mainNode )
         $pathParts = array( eZSys::storageDirectory(), $contentImageSubtree );
         if ( $pathString != '' )
         {
-            $pathParts[] = $pathString;
+            //Make sure that all folders are smaller than the FS limit
+            foreach ( explode( '/', $pathString ) as $folder )
+            {
+                $pathParts[] = self::trimToFileSystemFileName( $folder );
+            }
         }
         $pathParts[] = $attributeID . '-' . $attributeVersion . '-' . $attributeLanguage;
         $imagePath = implode( '/', $pathParts );
         return $imagePath;
     }
 
+    /**
+     * Trims a filename to a given limit (in bytes).
+     * It can be used to avoid file system limitations
+     *
+     * @param $longName
+     * @param int $limit in bytes
+     *
+     * @return string shortened name
+     */
+    public static function trimToFileSystemFileName( $longName, $limit = 200 )
+    {
+        if ( strlen( $longName ) <= $limit )
```

Comment:
What Yannick means is that the issue is the space we have in the database, and this is counted as non-multibyte. We want to truncate the string that (non-mb) length, not to its true length.